### PR TITLE
PaLM API LLM component

### PR DESCRIPTION
**Pull request by mdda**
This is a first draft of the the LLM component for the PaLM API (ref: #862)

Screenshot_2023-09-06_16-24-33

Could you have a look at the overall style? 

Once this part is satisfactory, I can do PaLM-Chat and PaLM-embeddings to match...

For the update to package.json : I wasn't sure how to make sure this installed when building overall, since it's required by langchain, but didn't seem to have installed itself.

I noticed that the 'json' field in the optional parameters don't seem to be properly editable - either on this component, or on the existing OpenAI one - so maybe that needs an additional fix.

### CODE REVIEW

**Comment by chungyau97** [comment]
Hi @mdda, 

Thanks for the PR, appreciate it a lot. 
Could you share some results [[screenshots](https://github.com/FlowiseAI/Flowise/pull/504#issuecomment-1627131466)/[short video](https://github.com/FlowiseAI/Flowise/pull/114)]?

File: packages/components/nodes/llms/GooglePaLM/GooglePaLM.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        this.icon = 'Google_PaLM_Logo.svg'
+        this.category = 'LLMs'
+        this.description = 'Wrapper around Google MakerSuite PaLM large language models'
+        this.baseClasses = [this.type, ...getBaseClasses(GooglePaLM)]
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            credentialNames: ['googleMakerSuite'],
+            description:
```
Comment (chungyau97) [inline packages/components/nodes/llms/GooglePaLM/GooglePaLM.ts:31]: Recommend moving credential's description into `packages/components/credentials/GoogleMakerSuite.credential.ts`

Examples:
1. [packages/components/credentials/ConfluenceApi.credential.ts](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/credentials/ConfluenceApi.credential.ts#L15)
2. [packages/components/credentials/FigmaApi.credential.ts](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/credentials/FigmaApi.credential.ts#L15)

**Comment by chungyau97** [comment]
Everything looks good to me, the commented code is due to we haven't support array format input field.

**Review by chungyau97 — APPROVED** [review]

**Comment by mdda** [comment]
Great!  

I'll have a go at the PaLM chat and PaLM embeddings next...

**Pull request merged**